fitbenchmarking/controllers/base_controller.py

```diff
@@ -180,6 +184,13 @@ def software(self):
                 self._software = self.__class__.__name__[:-10].lower()
         return self._software
 
+    def validate(self) -> None:
```

Comment:
I'm not sure I understand why this has 2 layers (`validate`, `_validate_jacobian`). Is it just pre-empting further validation?